src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UITests/Presentation/SampleChooserViewModel.cs

```diff
@@ -254,124 +254,145 @@ internal async Task RecordAllTests(CancellationToken ct, string screenShotPath =
-					var testQuery = from category in _categories
-									from sample in category.SamplesContent
-									where !sample.IgnoreInSnapshotTests
-									// where sample.ControlName.Equals("GridViewVerticalGrouped")
-									select new SampleInfo
-									{
-										Category = category,
-										Sample = sample,
-									};
-
-					Debug.Assert(
-						_firstTargetToRun.IsNullOrEmpty() || testQuery.Any(testInfo => testInfo.Matches(_firstTargetToRun)),
-						"First target to run must be either a Category or a Sample that is present in the app."
-					);
-
-					Debug.Assert(
-						_targetsToSkip.Where(target => !target.IsNullOrWhiteSpace()).None(target => target.Equals(_firstTargetToRun, StringComparison.OrdinalIgnoreCase)),
-						"First test to run cannot be skipped"
-					);
-
-					var tests = testQuery
-						.SkipWhile(testInfo => _firstTargetToRun.HasValue() && !testInfo.Matches(_firstTargetToRun))
-						.Where(testInfo => _targetsToSkip.None(testInfo.Matches))
-						.ToArray();
+				var testQuery = from category in _categories
+								from sample in category.SamplesContent
+								where !sample.IgnoreInSnapshotTests
+								// where sample.ControlName.Equals("GridViewVerticalGrouped")
+								select new SampleInfo
+								{
+									Category = category,
+									Sample = sample,
+								};
+
+				Debug.Assert(
+					_firstTargetToRun.IsNullOrEmpty() || testQuery.Any(testInfo => testInfo.Matches(_firstTargetToRun)),
+					"First target to run must be either a Category or a Sample that is present in the app."
+				);
+
+				Debug.Assert(
+					_targetsToSkip.Where(target => !target.IsNullOrWhiteSpace()).None(target => target.Equals(_firstTargetToRun, StringComparison.OrdinalIgnoreCase)),
+					"First test to run cannot be skipped"
+				);
+
+				var tests = testQuery
+					.SkipWhile(testInfo => _firstTargetToRun.HasValue() && !testInfo.Matches(_firstTargetToRun))
+					.Where(testInfo => _targetsToSkip.None(testInfo.Matches))
+					.ToArray();
+
+				if (this.Log().IsEnabled(LogLevel.Debug))
+				{
+					this.Log().Debug($"Generating tests for {tests.Count()} test in {folderName}");
+				}
 
-					if (this.Log().IsEnabled(LogLevel.Debug))
+				foreach (var sample in tests)
+				{
+					try
 					{
-						this.Log().Debug($"Generating tests for {tests.Count()} test in {folderName}");
-					}
+#if XAMARIN
+						var inactiveStats = Uno.UI.DataBinding.BinderReferenceHolder.GetInactiveViewReferencesStats();
+						var activeStats = Uno.UI.DataBinding.BinderReferenceHolder.GetReferenceStats();
+#endif
+
+						var fileName = $"{sample.Category.Category}-{sample.Sample.ControlName}.png";
 
-					foreach (var sample in tests)
-					{
 						try
 						{
-#if XAMARIN
-							var inactiveStats = Uno.UI.DataBinding.BinderReferenceHolder.GetInactiveViewReferencesStats();
-							var activeStats = Uno.UI.DataBinding.BinderReferenceHolder.GetReferenceStats();
-#endif
+							Console.WriteLine($"Creating control for {fileName}");
```

Comment:
Codacy found an issue: [Remove this logging statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6226524)